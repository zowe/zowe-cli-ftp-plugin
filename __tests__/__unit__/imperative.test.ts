/*
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 *
 */

describe("imperative config", () => {

    // Will fail if impertive config object is changed. This is a sanity/protection check to ensure that any
    // changes to the configuration document are intended (and the snapshot must be updated).
    it("should match the snapshot", () => {
        const config = require("../../src/imperative");
        expect(config).toMatchSnapshot();
    });
});
