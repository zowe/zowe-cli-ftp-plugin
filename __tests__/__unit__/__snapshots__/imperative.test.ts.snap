// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`imperative config should match the snapshot 1`] = `
Object {
  "commandModuleGlobs": Array [
    "**/cli/*/*.definition!(.d).*s",
  ],
  "name": "zos-ftp",
  "pluginAliases": Array [
    "zftp",
  ],
  "pluginHealthCheck": "./lib/healthCheck.handler",
  "pluginSummary": "z/OS Files and jobs via FTP",
  "productDisplayName": "z/OS FTP Plugin",
  "profiles": Array [
    Object {
      "createProfileExamples": Array [
        Object {
          "description": "Create a zftp profile called 'myprofile' with default settings (port, timeout, etc.) to connect with the host system 123.",
          "options": "myprofile -u ibmuser -p ibmp4ss -H sys123",
        },
      ],
      "schema": Object {
        "description": "Configuration profile for z/OS FTP",
        "optional": Array [
          "host",
          "port",
          "user",
          "password",
          "encoding",
        ],
        "properties": Object {
          "connectionTimeout": Object {
            "optionDefinition": Object {
              "aliases": Array [
                "ct",
              ],
              "defaultValue": 10000,
              "description": "How long (in milliseconds) to wait for the control connection to be established.",
              "group": "FTP Connection options",
              "name": "connection-timeout",
              "type": "number",
            },
            "type": "number",
          },
          "encoding": Object {
            "optionDefinition": Object {
              "aliases": Array [
                "ec",
              ],
              "defaultValue": null,
              "description": "The encoding for download and upload of z/OS data set.",
              "group": "FTP Connection options",
              "name": "encoding",
              "type": "string",
            },
            "type": "string",
          },
          "host": Object {
            "optionDefinition": Object {
              "aliases": Array [
                "H",
              ],
              "description": "The hostname or IP address of the z/OS server to connect to.",
              "group": "FTP Connection options",
              "name": "host",
              "type": "string",
            },
            "type": "string",
          },
          "password": Object {
            "optionDefinition": Object {
              "aliases": Array [
                "p",
                "pass",
                "pw",
              ],
              "description": "Password to authenticate to FTP.",
              "group": "FTP Connection options",
              "name": "password",
              "type": "string",
            },
            "secure": true,
            "type": "string",
          },
          "port": Object {
            "includeInTemplate": true,
            "optionDefinition": Object {
              "aliases": Array [
                "P",
              ],
              "defaultValue": 21,
              "description": "The port of the z/OS FTP server.",
              "group": "FTP Connection options",
              "name": "port",
              "type": "number",
            },
            "type": "number",
          },
          "rejectUnauthorized": Object {
            "optionDefinition": Object {
              "aliases": Array [
                "ru",
              ],
              "defaultValue": null,
              "description": "Reject self-signed certificates. Only specify this if you are connecting to a secure FTP instance.",
              "group": "TLS / Secure Connection options",
              "name": "reject-unauthorized",
              "type": "boolean",
            },
            "type": Array [
              "boolean",
              "null",
            ],
          },
          "secureFtp": Object {
            "includeInTemplate": true,
            "optionDefinition": Object {
              "defaultValue": true,
              "description": "Set to true for both control and data connection encryption, 'control' for control connection encryption only, or 'implicit' for implicitly encrypted control connection (this mode is deprecated in modern times, but usually uses port 990). Note: Unfortunately, this plugin's functionality only works with FTP and FTPS, not 'SFTP' which is FTP over SSH.",
              "group": "FTP Connection options",
              "name": "secure-ftp",
              "type": "boolean",
            },
            "type": Array [
              "boolean",
              "null",
            ],
          },
          "servername": Object {
            "optionDefinition": Object {
              "aliases": Array [
                "sn",
              ],
              "defaultValue": null,
              "description": "Server name for the SNI (Server Name Indication) TLS extension. Only specify if you are connecting securely",
              "group": "TLS / Secure Connection options",
              "name": "server-name",
              "type": "string",
            },
            "type": Array [
              "string",
              "null",
            ],
          },
          "user": Object {
            "optionDefinition": Object {
              "aliases": Array [
                "u",
              ],
              "description": "Username for authentication on z/OS",
              "group": "FTP Connection options",
              "name": "user",
              "type": "string",
            },
            "secure": true,
            "type": "string",
          },
        },
        "title": "Configuration profile for z/OS FTP",
        "type": "object",
      },
      "type": "zftp",
    },
  ],
  "rootCommandDescription": "Data set and job functionality via FTP. This functionality uses the open source zos-node-accessor package from IBM. Commands under this group require you to create a zftp profile before using them.
If you find this functionality useful, please consider setting up z/OSMF on your system to get improved stability and speed and more features (for example, issuing TSO and console commands) by using core Zowe CLI.",
}
`;
